src/app/common/use-wallet-type.ts

```diff
@@ -14,10 +14,10 @@ function isSoftwareWallet(walletType: WalletType) {
 
 type WalletTypeMap<T> = Record<WalletType, T>;
 
-function whenWalletFactory(walletType: WalletType) {
-  return <T>(walletTypeMap: WalletTypeMap<T>): T => {
-    if (isLedgerWallet(walletType)) return walletTypeMap.ledger;
-    if (isSoftwareWallet(walletType)) return walletTypeMap.software;
+function whenWallet(walletType: WalletType) {
+  return <T extends WalletTypeMap<unknown>>(walletTypeMap: T) => {
+    if (isLedgerWallet(walletType)) return walletTypeMap.ledger as T['ledger'];
```

Comment:
Would it be better to start using `hardware` everywhere, rather than `ledger` specific naming?

Reply:
Is the idea is that each hardware wallet might need to be handled differently?

Reply:
I've wondered this. If we can handle them the same, `hardware` is better, if not, we might need to consider them their own types. Really depends how the trezor app works.